torch/utils/hooks.py

```diff
@@ -34,3 +35,14 @@ def __enter__(self):
 
     def __exit__(self, type, value, tb):
         self.remove()
+
+
+def warn_if_has_hooks(tensor):
+    if tensor._backward_hooks:
+        for k in tensor._backward_hooks:
+            hook = tensor._backward_hooks[k]
+            if not hasattr(k, "_torch_unserializable"):
```

Comment:
Also, another problem is that when users add hooks via `tensor.add_hook`, what they get is a handle, which is not the hook object stored in `_backward_hooks`. So effectively there is no directly way to set `_torch_unserializable` on the hook. Maybe make that a method on the handle (or an attribute with a setter)?

Reply:
Hmm, my intention was originally that you mark the hook as unserializable when you attach it using `add_hook`, rather than post-facto from the handle. (If you had a decorator, would be `unserializable_hook(lambda t: my_hook_impl())`. Does that make sense?

Reply:
Ah yes it makes sense. Thanks!